src/App.tsx

```diff
@@ -1,24 +1,112 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
+
+import { ErrorNotification } from './components/ErrorNotification';
+import { FooterMenu } from './components/FooterMenu';
+import { ListOfTodos } from './components/ListOfTodos';
+import { Header } from './components/Header';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import { getTodos } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { CustomError } from './types/CustomError';
+import { ActiveTodoData } from './types/ActiveTodoData';
+
+import { USER_ID } from './utils/fetchClient';
+import { initData } from './constants/initData';
+import { useError } from './utils/useError';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initData.todos);
+  const [filter, setFilter] = useState<Filter>(initData.filter);
+  const [customError, setDelayError]
+    = useError(initData.customError);
+  const [activeTodoData, setActiveTodo]
+    = useState<ActiveTodoData>(initData.activeTodoData);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(initData.tempTodo);
+
+  const filterCallback = ((todo: Todo) => {
+    switch (filter) {
+      case Filter.Completed:
+        return todo.completed;
+      case Filter.Active:
+        return !todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setActiveTodo({
+          hasActiveTodo: todosFromServer.some(todo => !todo.completed),
+          activeLeft: todosFromServer.filter(todo => !todo.completed).length,
+        });
+      })
+      .catch(() => {
+        setDelayError(CustomError.update, 3000);
+      });
+  }, []);
+
+  useEffect(() => {
+    setActiveTodo({
+      hasActiveTodo: !!todos.some(todo => !todo.completed),
+      activeLeft: todos.filter(todo => !todo.completed).length,
+    });
+    setDelayError(CustomError.noError);
+  }, [todos]);
+
+  const filteredTodos = useMemo(
+    () => todos.filter(filterCallback),
```

Comment:
```
//[HINT] if you need to filter your items by selected filter - use switch/case statement in correct way.
// Perform the necessary filtering only on a case-by-case basis.

// GOOD

const filteredCards = useMemo(() => {
  switch (filter) {
    case CardFilters.All: {
      return cards;
    }

    case CardFilters.Active: {
      return cards.filter(({ deletedDate }) => !deletedDate);
    }

    case CardFilters.Deleted: {
      return cards.filter(({ deletedDate }) => Boolean(deletedDate));
    }

    default:
      return [];
  }
}, [cards, filter]);

// Don't filter your array right away with switch/case inside your callback.
// Because in case CardFilters.All your program just iterate an array and doesn't do any useful actions. So it's a wasting of memory and CPU (very bad)

// BAD

const filteredCards = useMemo(() => {
  return cards.filter(({ deletedDate }) => {
    switch (filter) {
      case CardFilters.All: {
        return true;
      }

      case CardFilters.Active: {
        return !deletedDate;
      }

      case CardFilters.Deleted: {
        return Boolean(deletedDate);
      }

      default:
        return false;
    }
  });
}, [cards, filter]);
```